common/functions/src/scalars/expressions/cast_with_type.rs

```diff
@@ -66,23 +69,27 @@ pub fn cast_column_field(
     column_with_field: &ColumnWithField,
     from_type: &DataTypeImpl,
     target_type: &DataTypeImpl,
+    func_ctx: &FunctionContext,
```

Comment:
How about putting `func_ctx` as the first argument?

Reply:
We can do this little refactor in another PR